src/ripple/ledger/impl/View.cpp

```diff
@@ -1264,22 +1064,29 @@ offerDelete (ApplyView& view,
     auto owner = sle->getAccountID  (sfAccount);
 
     // Detect legacy directories.
-    bool bOwnerNode = sle->isFieldPresent (sfOwnerNode);
     uint256 uDirectory = sle->getFieldH256 (sfBookDirectory);
 
-    TER terResult  = dirDelete (view, false, sle->getFieldU64 (sfOwnerNode),
-        keylet::ownerDir (owner), offerIndex, false, !bOwnerNode, j);
-    TER terResult2 = dirDelete (view, false, sle->getFieldU64 (sfBookNode),
-        keylet::page (uDirectory), offerIndex, true, false, j);
+    bool deletedOwner = view.dirRemove(
+        keylet::ownerDir(owner),
+        sle->getFieldU64(sfOwnerNode),
+        offerIndex,
+        false);
 
-    if (tesSUCCESS == terResult)
+    view.dirRemove(
+        keylet::page(uDirectory),
+        sle->getFieldU64(sfBookNode),
+        offerIndex,
+        false);
+
+    if (deletedOwner)
+    {
         adjustOwnerCount(view, view.peek(
             keylet::account(owner)), -1, j);
+    }
 
     view.erase(sle);
 
-    return (terResult == tesSUCCESS) ?
-        terResult2 : terResult;
+    return tesSUCCESS;
```

Comment:
It seems like this should return `tefBAD_LEDGER` if either of the removes weren't successful.